components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/UsageClient.java

```diff
@@ -0,0 +1,217 @@
+     */
+    public static boolean isDataPublishingEnabled() {
+        APIManagerAnalyticsConfiguration con = APIManagerAnalyticsConfiguration.getInstance();
+        return con.isAnalyticsEnabled();
+    }
+
+
+    /**
+     * Use to get instance of implementation class of the APIUsageStatisticsClient that is defined in the apim-manager.xml
+     *
+     * @return instance of a APIUsageStatisticsClient
+     * @throws APIMgtUsageQueryServiceClientException throws if instantiation problem occur
+     */
+    private static APIUsageStatisticsClient getStatisticClient(String user) throws APIMgtUsageQueryServiceClientException {
+
+        //read the api-manager.xml and get the Statistics class name
+        APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                .getAPIManagerConfiguration();
+        String className = config.getFirstProperty("StatisticClientProvider");
+
+        try {
+
+            //get the Class from the class name
+            Class statClass=APIUtil.getClassFromName(className);
+            //use the constructor and pass appropriate args to get a instance
+            if(user!=null) {
+                usageStatisticsClient = (APIUsageStatisticsClient) statClass.getConstructor(String.class)
+                        .newInstance(user);
+            }else{
+                usageStatisticsClient = (APIUsageStatisticsClient) statClass.getConstructor()
+                        .newInstance();
+            }
+
+        } catch (InstantiationException e) {
+            throw new APIMgtUsageQueryServiceClientException("Cannot instantiate Statistic Client class: " + className,
+                    e);
+        } catch (IllegalAccessException e) {
+            throw new APIMgtUsageQueryServiceClientException(
+                    "Cannot access the constructor in Statistic Client class: " + className, e);
+        } catch (InvocationTargetException e) {
+            throw new APIMgtUsageQueryServiceClientException("Error occurred while getting constructor");
+        } catch (NoSuchMethodException e) {
+            throw new APIMgtUsageQueryServiceClientException(
+                    "Cannot found expected constructor in Statistic Client class: " + className, e);
+        } catch (ClassNotFoundException e) {
+            throw new APIMgtUsageQueryServiceClientException("Cannot found the Statistic Client class: " + className,
+                    e);
+        }
+
+        return usageStatisticsClient;
+    }
+
+
+    /**
+     * Get the Subscriber count and information related to the APIs
+     *
+     * @param loggedUser user of the current session
+     * @return return list of SubscriberCountByAPIs objects. which contain the list of apis and related subscriber counts
+     * @throws APIManagementException throws exception if error occur
+     */
+    public static List<SubscriberCountByAPIs> getSubscriberCountByAPIs(String loggedUser,boolean isAllStatistics)
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/e34c4353fe49f6ce3ff615cb8ea5074d314f5e35